### ISSUE TITLE
[Bug]: Create HNSW index on binary vectors should not allowed

### ISSUE THREAD

**Issue by bigsheeper** [milvus-io/milvus#23497]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master/ maybe also 2.2.x
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

I performed an integration test to create an HNSW index on binary vectors. However, no errors occurred, which violates the similarity metrics rule mentioned in https://milvus.io/docs/metric.md#binary. 

I think Proxy should make a check about this @longjiquan ,
and Knowhere should also make a check @liliu-z. 

You can use the test code as a reference: https://github.com/bigsheeper/milvus/blob/2304-get-vectors-3/tests/integration/get_vector_test.go

**Comment by bigsheeper** [comment]
![img_v2_710acd68-40c2-4416-998d-1e10b2f4231g](https://user-images.githubusercontent.com/42060877/232725341-90769b25-4040-4a54-a5cc-796484f9a274.jpg)

![img_v2_4e0d975f-3ed2-4af0-92b6-49cdfd147b5g](https://user-images.githubusercontent.com/42060877/232725648-f42d29a8-43f0-4711-a963-4439cd6268b4.jpg)


pymilvus made a check. Please help to make a verify about GO SDK and Java SDK @congqixia @yhmo

**Comment by longjiquan** [comment]
I don't think it's good to check it on the client side.

**Comment by longjiquan** [comment]
Btw, anyone knows why we add the check logic in the pymilvus?

**Comment by liliu-z** [comment]
good catch, should legacy problem

**Comment by cydrain** [comment]
Milvus already has this check for index type and metric type matching.
When set metric type to "HAMMING" and index type to "HNSW", and comment the check in pymilvus,
Milvus will also report following error:
```
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=invalid index params: map[M:8 dim:128 efConstruction:200 index_type:HNSW metric_type:HAMMING])>
```
this error is from 
![Screenshot from 2023-04-21 14-22-23](https://user-images.githubusercontent.com/3992404/233556792-34bde91a-2f9c-44b5-876f-98eeaf02dec7.png)

**Comment by czs007** [comment]
keep it

**Comment by cydrain** [comment]
@bigsheeper

**Comment by longjiquan** [comment]
/unassign
/assign @bigsheeper 
already fixed in #23856 and #23903
